src/main/java/com/minecolonies/coremod/colony/ColonyView.java

```diff
@@ -254,6 +347,7 @@ public CitizenDataView getCitizen(final int id)
      * @param isNewSubscription Whether this is a new subscription of not.
      * @return null == no response.
      */
+    @Nullable
     public IMessage handleColonyViewMessage(@NotNull final ByteBuf buf, final boolean isNewSubscription)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Provide multiple methods instead of using "isNewSubscription" to determine which action to take. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2301)